packages/terra-functional-testing/src/webpack-server/webpack-server.js

```diff
@@ -73,8 +76,30 @@ class WebpackServer {
         if (stats.hasErrors()) {
           logger.error('Webpack compiled with errors.');
           reject();
-        } else {
+        } else if (!this.gridUrl) {
           resolve();
+        } else {
+          let response = '';
+          http.get(this.gridUrl, (res) => {
+            const { statusCode } = res;
+            if (statusCode >= 200 && statusCode <= 299) {
+              res.on('data', (chunk) => {
+                response += chunk;
+              });
+              res.on('end', () => {
+                if (!res.complete) throw new SevereServiceError(`${this.gridUrl} connection was terminated while the message was still being sent`);
+                if (JSON.parse(response).value.ready) {
+                  resolve();
+                } else {
+                  throw new SevereServiceError(`${this.gridUrl} failed to return a ready response. Check to ensure the selenium grid is stable`);
+                }
+              });
+            } else {
+              throw new SevereServiceError(`Url ${this.gridUrl} returns status code of ${statusCode}.`);
```

Comment:
In the case of a failure do we need to call resume() on the response to free up memory as indicated by its [documentation](https://nodejs.org/api/http.html#http_http_get_url_options_callback)?

Reply:
Also for failure cases don't we need to reject the promise?

Reply:
So added resume and rejects here https://github.com/cerner/terra-toolkit/pull/679/commits/900b6e80e6616822114af05e0ef367660049fb53#diff-73928b81969e1fbd0b7fed98d5dc39d99e26fd5d75ace8ef586e3a4b2e25e0ffR101

the `resume()` seems to mostly just free up the memory earlier but using the above documentation and this [tutorial](https://www.digitalocean.com/community/tutorials/how-to-create-an-http-client-with-core-http-in-node-js) it looks like we can just place in the error message with the status code reply and good. Feel free to comment and offer suggestions for better implementation.